### PR TITLE
packages/cli,core-api: move config reading and loading into separate packages

### PR DESCRIPTION
**Pull request by Rugvip**
Tried out some ways to not have to split these into more packages, but that all ended up being a pain and require more build tooling, so ended up with two new packages from config reading/loading.

Also a bit of a refactor to make the reader more extensible, but in the end I think it's best to just have a single reader implementation that's used everywhere.

### CODE REVIEW

File: packages/config-loader/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# @backstage/config-loader
+
+This package provides config loading functionality used by the backend, and CLI.
+
+## Installation
+
+Do not install this package directly, it is an internal packages used by [@backstage/cli](https://www.npmjs.com/package/@backstage/cli), and [@backstage/backend-common](https://www.npmjs.com/package/@backstage/backend-common). Depend on either of those instead.
```
Comment (freben) [inline packages/config-loader/README.md:7]: `internal package`

File: packages/config-loader/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "@backstage/config-loader",
+  "description": "Config API used by Backstage core, backend, and CLI",
```
Comment (freben) [inline packages/config-loader/package.json:3]: not core, right?

File: packages/config-loader/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  "version": "0.1.1-alpha.6",
+  "private": false,
+  "publishConfig": {
+    "access": "public"
+  },
+  "homepage": "https://backstage.io",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/spotify/backstage",
+    "directory": "packages/config"
```
Comment (freben) [inline packages/config-loader/package.json:13]: config-loader

File: packages/config-loader/src/loader.ts [hunk 4]
```diff
@@ -14,22 +14,25 @@
...
 export async function loadConfig(
   options: LoadConfigOptions = {},
 ): Promise<AppConfig[]> {
   // TODO: We'll want this to be a bit more elaborate, probably adding configs for
   //       specific env, and maybe local config for plugins.
-  const { configPath = paths.resolveTargetRoot('app-config.yaml') } = options;
+  let { configPath } = options;
+  if (!configPath) {
+    configPath = resolvePath(
+      findRootPath(fs.realpathSync(process.cwd())),
```
Comment (freben) [inline packages/config-loader/src/loader.ts:32]: I was a bit surprised to find use of cwd here, not saying it's wrong. I just have been bitten by that before in cases where you trigger the running of a tool from a shell in a dir way outside of the path that it then actually targets.

File: packages/config/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# @backstage/config
+
+This package provides a config API used by Backstage core, backend, and CLI.
+
+## Installation
+
+Do not install this package directly, it is an internal packages used by [@backstage/core](https://www.npmjs.com/package/@backstage/core), [@backstage/cli](https://www.npmjs.com/package/@backstage/cli), and [@backstage/backend-common](https://www.npmjs.com/package/@backstage/backend-common). Depend on either of those instead.
```
Comment (freben) [inline packages/config/README.md:7]: `internal package`

**Comment by Rugvip** [comment]
Think we'll have to point the config packages main straight to `src`, which means we can't use them in the backend yet

**Review by freben — APPROVED** [review]

**Pull request merged**